torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -289,6 +289,7 @@ void GpuLower::lower(Fusion* fusion, DataType index_type) {
 
   doubleBufferInfo().build(fusion_);
 
+  compute_at_map_->allocateIndexVariables();
```

Comment:
Is there a particular reason this is built so late and not just right next to the compute at map? If yes, we may want to make it a separate class from compute at map.

Reply:
Will make it a separate class in a follow up, since this info actually rely on pre-built ca map.